### ISSUE TITLE
an elegant way to check for time ranges in rules

### ISSUE THREAD

**Issue by jimtng** [boc-tothefuture/openhab-jruby#41]
It would be nice if we could check for time ranges inside our rule code similar to the `between` rule block / property. It is not an uncommon task to do.

@boc-tothefuture suggested having a function called `between` to convert a string range into a range of TimeOfDay, which would be great.

```
case TimeOfDay.now
when between('8:00'...'12:00') then x
when between('12:00'..'16:00') then x
else then y
```

```
between('8:00'..'12:00').cover? TimeOfDay.now
```

Where would such 'helper' functions live? Such would be a place also for the store/restore states helper function mentioned in https://github.com/boc-tothefuture/openhab-jruby/issues/31

**Comment by boc-tothefuture** [comment]
The between function should also replace how it is currently handled in the rule block which prepends the Range class to handle time of day specifically.

**Comment by jimtng** [comment]
@boc-tothefuture would you be able to implement this?

**Comment by boc-tothefuture** [comment]
Yes, happy to.  I am working still on the threading/context issues at present which are challenging.. But critical to having a solid base.

**Comment by boc-tothefuture** [comment]
Addressed by #57
